nodejs/week2/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,107 @@
+const { query, request } = require("express");
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+//...............Respond with the json for the meal with the corresponding id
+
+router.get("/:id", async (request, response) => {
+
+  try {
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+    
+    const amountOfMeals = meals.length;
+    const realId = parseInt(request.params.id) - 1;
+    const checkId = parseFloat(request.params.id);
+    
+    //...What if there is no meal with the requested id
+
+    if (!isNaN(request.params.id) && Number.isInteger(checkId) && 0 <= realId && realId < amountOfMeals) {
+      correspondAnswer = meals[realId];
```

Comment:
The correct meal is not necessarily at the id index of the list. For example, you could have:

    meals = [{id: 1}, {id: 5}]

Reply:
Would you descripe me more. Did I do wrong?

Reply:
I make it correct. Thanks